### DISCUSSION TITLE
Toast is throwing a System.Runtime.InteropServices.COMException

### DISCUSSION THREAD

**Discussion by Blackswater** [CommunityToolkit/Maui#1527] [Q&A]
Hello all,
I have a Problem and I don't know really how to solve it.
I have a quite simple Application, it is just a Page in xaml with a Button. The Button as a Clicked handler, which is showing a Toast. 
This is how I create the Toast
private async void Button_Clicked(object sender, EventArgs e)
{
    var toast = Toast.Make("Test");
    await toast.Show();
}

the Problem is that the Show method is throwing the System.Runtime.InteropServices.COMException.

What am I doing wrong?

I am on .Net8 and updated all Nuget Packages.

The CommunityToolkitSample solution works fine for me, so I think it is something with my solution. But I don't know what it could be.

**Comment by GeorgeLeithead** [comment]
Are you using:

using CommunityToolkit.Maui.Alerts;

Then something like:
	async void TapGestureRecognizer_Tapped(object sender, EventArgs e)
	{
		var cts = new CancellationTokenSource(TimeSpan.FromSeconds(5));
		await ShowToastGestureMessage("AvatarView Tap Gesture Recognizer, tapped.", cts.Token);
	}

	static Task ShowToastGestureMessage(string message, CancellationToken token)
	{
		Core.IToast toast = Toast.Make(message, Core.ToastDuration.Short);
		return toast.Show(token);
	}

**Reply by Blackswater** [reply]
I will try this out.

**Reply by Blackswater** [reply]
This does not work in my solution. Trying to prepare a sample solution for this

**Comment by GeorgeLeithead** [comment]
Can you attach a small REPO, so that everyone/anyone can review and see what can be done to resolve/help?

**Reply by bijington** [reply]
Yes I would recommend reproducing it in a sample app and then opening a bug report

**Reply by Blackswater** [reply]
I am trying to reproduce it with a sample app

**Reply by Blackswater** [reply]
@GeorgeLeithead  @bijington 
this is how you can simply reproduce it.

1. Create a new Solution
2. Add the Nuget Package
3. Create a Toast
4. Change the commandName to "Project" in the launcherSettings.json
5. add in the csproj this: 		`<WindowsPackageType Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">None</WindowsPackageType>`

The Toast will not be shown anymore.

**Reply by bijington** [reply]
Why are you changing the settings in points 4 and 5? 

I'm just guessing now but I suspect that the AppNotification API that the Windows implementation is using relies on something that doesn't play well with those settings

**Reply by Blackswater** [reply]
I need to do this, because I need to ship the App via a MSI and not over Msix. The MSI is builded via WIX. If you have any other Idea, how I could ship the App with Msix and WIX, let me know

**Reply by bijington** [reply]
That's outside of my area of knowledge. What I would suggest doing is trying to find out whether there is a dependency which is being shipped through the MSIX approach which you are not including in your MSI. It might be a case of building the MSIX and comparing what gets installed vs your current install. Or try running something like FUSLOGVW (https://learn.microsoft.com/en-us/dotnet/framework/tools/fuslogvw-exe-assembly-binding-log-viewer) and see if it detects something isn't being loaded that you rely on.

**Reply by Blackswater** [reply]
Thanks for your help. I need to look around to see, what I can do.

**Comment by Blackswater** [comment]
Maybe worth to mention is that, I have in the launcherSettings.json this:
{
  "profiles": {
    "Windows Machine": {
      "commandName": "Project",
      "nativeDebugging": false
    }
  }
}

So I don't build it as MsixPackage

**Comment by CliffAgius** [comment]
Closed as answered, if closed in error please let us know to re-open.
